src/test/java/guitests/GuiRobot.java

```
@@ -37,4 +40,39 @@ public void pauseForHuman(int duration) {
 
         sleep(duration);
     }
+
+    /**
+     * Wait for {@code event} to be true.
+     *
+     * @param timeOut in milliseconds
+     * @throws EventTimeoutException if the time taken exceeds {@code timeOut}.
+     */
+    public void waitForEvent(BooleanSupplier event, int timeOut) {
+        int timePassed = 0;
+        final int retryInterval = 50;
+
+        while (!event.getAsBoolean()) {
+            sleep(retryInterval);
+            timePassed += retryInterval;
+
+            if (timePassed >= timeOut) {
+                throw new EventTimeoutException();
+            }
+        }
+    }
+
+    /**
+     * Returns true if the window with {@code stageTitle} is currently open.
+     */
+    public boolean isWindowActive(String stageTitle) {
+        return listTargetWindows().stream()
+                .filter(window -> window instanceof Stage && ((Stage) window).getTitle().equals(stageTitle))
+                .count() == 1;
```

Should we be more lax by just checking for "at least one"? i.e. `count() > 1'? Otherwise if there are two windows with the same title it is still technically wrong to return false, I think. (Or maybe throw an exception in that case)